examples/python-cicd-with-cloudbuilder/tests/test_my_module.py

```diff
@@ -0,0 +1,45 @@
+"""
+   Copyright 2019 Google LLC
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+"""
+from my_module import *
+import pytest
+
+
+class TestAdd(object):
+  def test_add_basic(selfs):
+    assert add(1,2) == 3
+
+  def test_add_zero(selfs):
+    assert add(0,3) == 3
+
+  def test_add_negative(selfs):
+    assert add(1,-2) == -1
+
+
+class TestSquare(object):
+  def test_square_positive(self):
+    assert square(2) == 4
+
+  def test_square_zero(self):
+    assert square(0) == 0
+
+  def test_square_negative(self):
+    assert square(-1) == 1
+
+class TestLogTransform(object):
```

Comment:
A couple more tests would be good for log_transform:
1. Testing it with and without the default constant value being used.
1. Testing to make sure the right exception is raised if constant is <= 0.

Reply:
Added these.  I wasn't really building this to be an example of how to write good unit tests (that's a book or a college course) but I agree with you. Further I think that bad/incomplete tests could take away credibility from the example.